mesonbuild/dependencies/detect.py

```diff
@@ -41,32 +41,34 @@ def __contains__(self, key: object) -> bool:
 def get_dep_identifier(name: str, kwargs: T.Dict[str, T.Any]) -> 'TV_DepID':
     identifier: 'TV_DepID' = (('name', name), )
     from ..interpreter.type_checking import DEPENDENCY_KWS
-    nkwargs = {k.name: k.default for k in DEPENDENCY_KWS}
-    nkwargs.update(kwargs)
 
     assert len(DEPENDENCY_KWS) == 20, \
            'Extra kwargs have been added to dependency(), please review if it makes sense to handle it here'
-    for key, value in nkwargs.items():
-        # 'version' is irrelevant for caching; the caller must check version matches
-        # 'native' is handled above with `for_machine`
-        # 'required' is irrelevant for caching; the caller handles it separately
-        # 'fallback' and 'allow_fallback' is not part of the cache because,
-        #     once a dependency has been found through a fallback, it should
-        #     be used for the rest of the Meson run.
-        # 'default_options' is only used in fallback case
-        # 'not_found_message' has no impact on the dependency lookup
-        # 'include_type' is handled after the dependency lookup
-        if key in {'version', 'native', 'required', 'fallback', 'allow_fallback', 'default_options',
-                   'not_found_message', 'include_type'}:
-            continue
+    # 'version' is irrelevant for caching; the caller must check version matches
+    # 'native' is handled above with `for_machine`
+    # 'required' is irrelevant for caching; the caller handles it separately
+    # 'fallback' and 'allow_fallback' is not part of the cache because,
+    #     once a dependency has been found through a fallback, it should
+    #     be used for the rest of the Meson run.
+    # 'default_options' is only used in fallback case
+    # 'not_found_message' has no impact on the dependency lookup
+    # 'include_type' is handled after the dependency lookup
+    # 'method' does not matter, we really only want one kind of a dependency
+    invalid_keys = {
+        'version', 'native', 'required', 'fallback', 'allow_fallback',
+        'default_options', 'not_found_message', 'include_type'}
+    valid_keys = [n for n in DEPENDENCY_KWS
+                  if n.name not in invalid_keys]
+    for dep in valid_keys:
```

Comment:
Why don't you like the `continue`? In this form, the list of keys needs to be iterated twice, which seems wasteful.

I would have written:
```python
for dep in DEPENDECY_KWS:
    if dep.name in invalid_keys:
        continue
    ...
```

Reply:
Because this simplifies the loop body, and I generally prefer that. I don't know that it really matters either way though?

Reply:
No, it does not matter. I was just curious because you mentioned the elimination of the `continue` in the commit message as it was something important. I prefer the code as I wrote it before, but it is mostly personal preference, I guess.